File: fe/lib/models/aimodels/CVData.dart
import 'package:ai_cv_generator/models/aimodels/AIEmployment.dart';
import 'package:ai_cv_generator/models/aimodels/AILink.dart';
import 'package:ai_cv_generator/models/aimodels/AIQualification.dart';
import 'package:ai_cv_generator/models/aimodels/AIReference.dart';
import 'package:ai_cv_generator/models/aimodels/AISkill.dart';
import 'package:json_annotation/json_annotation.dart';

part 'CVData.g.dart';

@JsonSerializable()
class CVData {
  String? firstname;
  String? lastname;
  String? email;
  String? phoneNumber;
  String? location; 
  String? description;
  List<AIEmployment>? employmenthistory;
  List<AIQualification>? qualifications;
  List<AILink>? links;
  List<AIReference>? references;
  List<AISkill>? skills;

  CVData({
    this.firstname,
    this.lastname,
    this.email,
    this.phoneNumber,
    this.location,
    this.description,
    this.employmenthistory,
    this.qualifications,
    this.links,
    this.references,
    this.skills
  });

  factory CVData.fromJson(Map<String,dynamic> json) => _$CVDataFromJson(json);

    Map<String,dynamic> toJson() => _$CVDataToJson(this);
}

File: ai-cv-generator-backend/src/main/java/com/revolvingSolutions/aicvgeneratorbackend/model/aimodels/CVData.java
package com.revolvingSolutions.aicvgeneratorbackend.model.aimodels;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class CVData {
    private String firstname;
    private String lastname;
    private String email;
    private String phoneNumber;
    private String location;
    private String description;
    private List<AIEmployment> employmenthistory;
    private List<AIQualification> qualifications;
    private List<AILink> links;
    private List<AIReference> references;
    private List<AISkill> skills;
}
